mod/consensus-types/pkg/types/deposit.go

```diff
@@ -186,6 +186,14 @@ func (d *Deposit) GetTree() (*fastssz.Node, error) {
 /* -------------------------------------------------------------------------- */
 /*                             Getters and Setters                            */
 /* -------------------------------------------------------------------------- */
+// Equals returns true if the Deposit is equal to the other.
+func (d *Deposit) Equals(rhs *Deposit) bool {
+	return d.Pubkey == rhs.Pubkey &&
+		d.Credentials == rhs.Credentials &&
+		d.Amount == rhs.Amount &&
+		d.Signature == rhs.Signature &&
+		d.Index == rhs.Index
```

Comment:
for the fields that are special types, should we use `bytes.Equal` or the type's `Equals` method?

Reply:
I would say their equals, but we should be good now